hack/windows/Prepare-Node.ps1

```diff
@@ -112,18 +139,25 @@ Remove-Item -Force .\nssm.zip
 
 $env:path += ";$global:NssmInstallDirectory"
 $newPath = "$global:NssmInstallDirectory;" +
-        [Environment]::GetEnvironmentVariable("PATH", [EnvironmentVariableTarget]::Machine)
+[Environment]::GetEnvironmentVariable("PATH", [EnvironmentVariableTarget]::Machine)
 
 [Environment]::SetEnvironmentVariable("PATH", $newPath, [EnvironmentVariableTarget]::Machine)
 
 Write-Host "Registering kubelet service"
 nssm install kubelet $global:Powershell $global:PowershellArgs $global:StartKubeletScript
-nssm set kubelet DependOnService docker
+
+if ($ContainerRuntime -eq "Docker") {
+    nssm set kubelet DependOnService docker
+} elseif ($ContainerRuntime -eq "containerd") {
+    nssm set kubelet DependOnService containerd
+}
 
 New-NetFirewallRule -Name kubelet -DisplayName 'kubelet' -Enabled True -Direction Inbound -Protocol TCP -Action Allow -LocalPort 10250
 
-# Create netadapter for kube-proxy, the default full name of the adapter is "vEthernet (HNS Internal NIC)"
-& ./Prepare-ServiceInterface.ps1
+if ($InstallKubeProxy) {
+    # Create netadapter for kube-proxy, the default full name of the adapter is "vEthernet (HNS Internal NIC)"
+    & ./Prepare-ServiceInterface.ps1
```

Comment:
C:\k\Prepare-ServiceInterface.ps1 -InterfaceAlias "HNS Internal NIC" ?

Reply:
I have removed that from the Start Kubelet script and kept it here dependant on InstallKubeProxy param......HNS INTERNAL NIC is default value of InterfaceAlias , so we don't need to specify that